conn.go

```diff
@@ -457,3 +497,18 @@ func (l *Conn) reader() {
 		}
 	}
 }
+
+// a variation of net.SplitHostPort which returns string as port if not specified.
+func splitHostAndPort(hostport string) (string, string, error) {
+
+	if hostport == "" {
+		return "", "", errors.New("Empty hostport provided")
+	}
+
+	// make golang < 1.8 compatible
+	i := strings.LastIndexByte(hostport, ':')
```

Comment:
this fails on go 1.4: https://travis-ci.org/go-ldap/ldap/jobs/433990177

Reply:
fixed. The build for 1.9  failed, but I believe it's not related to my changes.